File: Recursion/Recursion.java

import java.math.BigInteger;
import java.util.*;

class Rec {
    public static void main(String []args){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        BigInteger number = BigInteger.valueOf(n);
        long before = System.currentTimeMillis();
        BigInteger result = fact(number);
        long after = System.currentTimeMillis();
        System.out.println("Elapsed Time: "+(after - before)/1000);
        // System.out.println("Result: "+result);
    }

    public static BigInteger fact(BigInteger number){
        if(number.compareTo(BigInteger.ONE)==0){
            return BigInteger.ONE;
        }else{
            return number.multiply(fact(number.subtract(BigInteger.ONE)));
        }
    }
}

File: Recursion/Recursion.go
package main

import (
	"fmt"
	"time"
	"math/big"
)

func main(){
	var input int64
	fmt.Scanln(&input)
	newBigInt := big.NewInt(input)
	var before = makeTimestamp()
	fact(newBigInt)
	var after = makeTimestamp()
	// fmt.Println("Result: ",result)	
	var elapsedTime = after - before
	fmt.Println("Elapsed Time: ",elapsedTime/1000)
}

func fact(bigInt *big.Int) *big.Int {
	if(bigInt.Cmp(big.NewInt(1))==0) {
		return big.NewInt(1)
	}else{
		c:= big.NewInt(1)
		c= c.Sub(bigInt,big.NewInt(1))
			return bigInt.Mul(bigInt,fact(c))
	}
}

func makeTimestamp() int64 {
    return time.Now().UnixNano() / int64(time.Millisecond)
}